R/verb-copy-to.R

```diff
@@ -166,40 +166,20 @@ op_vars.lazy_values_query <- function(op) {
   colnames(op$x)
 }
 
-sql_values <- function(con, df, lvl = 0, ...) {
+sql_values_subquery <- function(con, df, lvl = 0, ...) {
   check_dots_empty()
-  UseMethod("sql_values")
+  UseMethod("sql_values_subquery")
 }
 
 #' @export
-sql_values.DBIConnection <- function(con, df, lvl = 0, ...) {
-  sql_values_clause(con, df, lvl = lvl)
+sql_values_subquery.DBIConnection <- function(con, df, lvl = 0, ...) {
+  sql_values_subquery_default(con, df, lvl = lvl, row = FALSE, derived = FALSE)
 }
 
-#' @export
-sql_values.SQLiteConnection <- function(con, df, lvl = 0, ...) {
-  needs_escape <- purrr::map_lgl(df, ~ is(.x, "Date") || inherits(.x, "POSIXct"))
-  purrr::modify_if(df, needs_escape, ~ escape(.x, con = con, parens = FALSE, collapse = NULL)) %>%
-    sql_values_clause(con = con, lvl = lvl)
-}
-
-sql_values_clause <- function(con, df, row = FALSE, derived = FALSE, lvl = 0) {
+sql_values_subquery_default <- function(con, df, lvl, row, derived) {
```

Comment:
There are valid reasons to avoid exporting this machinery. Should we make it as easy as possible for third-party packages to vendor this code, e.g. by putting it into a separate file and mentioning a "do whatever you want" license?

Reply:
Rather not yet. It uses another unexported function `sql_format_clauses()`. So, we would also have to export this function. But then we should also export `sql_clause()` and maybe some others. Would be nice if we could keep them internal for now.

Reply:
I'm all for keeping them internal, just collect in one place so that implementers can grab them. Not important though.